File: PYTHON/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  silnia.py
#  

def silnia_it(n):
    # 0! = 1
    # 1! = 1
    # 4! = 24
    wynik = 1
    for i in range(1, n+1):
        print(wynik)
        wynik = i * wynik
    return wynik

def main(args):
    n = int(input("Podaj liczbę: "))
    print("{}! = {}".format(n, silnia_it(n)))
    return 0
    
if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: CPP/silnia.cpp
/*
 * silnia.cpp
 */

#include <iostream>
using namespace std;

int silnia_re(int n) {
    if (n == 0) return 1;
    else return n*silnia_re(n-1);

}

int main(int argc, char **argv){
    int n;
    cout << "Podaj liczbę: "; cin >> n;
    cout <<"Wynik: " << silnia_re(n) << endl;
    return 0; 
    
}
    
